src/Kiota.Builder/Writers/Python/PythonConventionService.cs

```diff
@@ -0,0 +1,171 @@
+        var parentBlock = targetElement.GetImmediateParentOfType<IBlock>();
+        if(parentBlock != null) {
+            var aliasedUsing = parentBlock.Usings
+                                                .FirstOrDefault(x => !x.IsExternal &&
+                                                                x.Declaration.TypeDefinition == targetType.TypeDefinition &&
+                                                                !string.IsNullOrEmpty(x.Alias));
+            return aliasedUsing?.Alias;
+        }
+        return null;
+    }
+    public override string GetTypeString(CodeTypeBase code, CodeElement targetElement, bool includeCollectionInformation = true, LanguageWriter writer = null) {
+        if(code is null)
+            return null;
+        var collectionPrefix = code.CollectionKind == CodeTypeCollectionKind.None && includeCollectionInformation ? string.Empty : "List[";
+        var collectionSuffix = code.CollectionKind == CodeTypeCollectionKind.None && includeCollectionInformation ? string.Empty : "]";
+        if(code is CodeUnionType currentUnion && currentUnion.Types.Any())
+            return currentUnion.Types.Select(x => GetTypeString(x, targetElement, true, writer)).Aggregate((x, y) => $"Union[{x}, {y.ToFirstCharacterLowerCase()}]");
+        else if(code is CodeType currentType) {
+            var typeName = GetTypeAlias(currentType, targetElement) ?? TranslateType(currentType);
+            if (TypeExistInSameClassAsTarget(code, targetElement))
+                typeName = targetElement.Parent.Name.ToFirstCharacterUpperCase();
+            if (code.ActionOf)
+                return WriteInlineDeclaration(currentType, targetElement, writer);
+            else
+                return $"{collectionPrefix}{typeName}{collectionSuffix}";
+        }
+        else throw new InvalidOperationException($"type of type {code.GetType()} is unknown");
+    }
+    #pragma warning restore CA1822 // Method should be static
+    internal static string RemoveInvalidDescriptionCharacters(string originalDescription) => originalDescription?.Replace("\\", "/");
+    public override string TranslateType(CodeType type) {
+        if (type.IsExternal)
+            return TranslateExternalType(type);     
+        else
+            return TranslateInternalType(type);
+    }
+    private static string TranslateExternalType(CodeType type) {
+        return type.Name switch  {
+                "String" or "string" => "str",
+                "integer" or "int32" or "int64" or "byte" or "sbyte" => "int",
+                "decimal" or "double" => "float",
+                "Binary" or "binary" => "bytes",
+                "void" => "None",
+                "DateTimeOffset" => "datetime",
+                "boolean" => "bool",
+                "Object" or "object" or "float" or "bytes" or "datetime" or "timespan" => type.Name,
+                _ => type.Name.ToFirstCharacterUpperCase().Replace("IParseNode", "ParseNode") ?? "object",
```

Comment:
this replace should be done in the refiner and removed from here

Reply:
Resolved in https://github.com/microsoft/kiota/pull/1200/commits/d9b3b56f2d85192d077a1adac758a45cbb42f89c